packages/block-editor/src/store/selectors.js

```diff
@@ -1284,6 +1284,45 @@ export const hasInserterItems = createSelector(
 	],
 );
 
+/**
+ * Determines whether there is only one item that may be inserted.
+ *
+ * @param {Object}  state        Editor state.
+ * @param {?string} rootClientId Optional root client ID of block list.
+ *
+ * @return {boolean} True if there is one item available, false if zero or more than one.
+ */
+export const hasOnlyOneAllowedInserterItem = ( state, rootClientId = null ) => {
+	if ( rootClientId ) {
+		const parentBlockListSettings = getBlockListSettings( state, rootClientId );
+		return get( parentBlockListSettings, [ 'allowedBlocks', 'length' ], 0 ) === 1;
+	}
+
+	return false;
+};
+
+/**
+ * Gets the name of the only item that may be inserted.
+ *
+ * @param {Object}  state        Editor state.
+ * @param {?string} rootClientId Optional root client ID of block list.
+ *
+ * @return {string?} The name of the allowed block or null.
+ */
+export const getTheOnlyAllowedItem = ( state, rootClientId = null ) => {
+	if ( ! rootClientId ) {
+		return null;
+	}
+	if ( ! hasOnlyOneAllowedInserterItem( state, rootClientId ) ) {
+		return null;
+	}
+
+	const parentBlockListSettings = getBlockListSettings( state, rootClientId );
+	const name = get( parentBlockListSettings, [ 'allowedBlocks', 0 ], 0 );
+
+	return name;
+};
```

Comment:
A request I should've made earlier, but it's a quick one: could you prefix these selectors with `__unstable` or `__experimental`?

Reply:
and done